internal/test/caser.go

```diff
@@ -0,0 +1,8 @@
+package test
+
+type Caser interface {
+	Name() string
+	Args() []interface{}
+	Fields() []interface{}
+	Check() func(...interface{}) error
```

Comment:
User need to do the type casting when implementing test case with this framework?

Reply:
Yes, it is.
we don't want to cast every time, so it might be good to do it on the frame side